admin/class-yoast-network-admin.php

```diff
@@ -0,0 +1,299 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internals
+ */
+
+/**
+ * Multisite utility class for network admin functionality.
+ */
+class Yoast_Network_Admin implements WPSEO_WordPress_Integration, WPSEO_WordPress_AJAX_Integration {
+
+	/**
+	 * Action identifier for updating plugin network options.
+	 */
+	const UPDATE_OPTIONS_ACTION = 'yoast_handle_network_options';
+
+	/**
+	 * Action identifier for restoring a site.
+	 */
+	const RESTORE_SITE_ACTION = 'yoast_restore_site';
+
+	/**
+	 * Gets the available sites as choices, e.g. for a dropdown.
+	 *
+	 * @param bool $include_empty Optional. Whether to include an initial placeholder choice.
+	 *
+	 * @return array Choices as $site_id => $site_label pairs.
+	 */
+	public function get_site_choices( $include_empty = false ) {
+		$choices = array();
+
+		if ( $include_empty ) {
+			$choices['-'] = __( 'None', 'wordpress-seo' );
+		}
+
+		$sites = get_sites( array( 'deleted' => 0 ) );
+		foreach ( $sites as $site ) {
+			$choices[ $site->blog_id ] = $site->blog_id . ': ' . $site->domain . $site->path;
+
+			$site_states = $this->get_site_states( $site );
+			if ( ! empty( $site_states ) ) {
+				$choices[ $site->blog_id ] .= ' [' . implode( ', ', $site_states ) . ']';
+			}
+		}
+
+		return $choices;
+	}
+
+	/**
+	 * Gets the states of a site.
+	 *
+	 * @param WP_Site $site Site object.
+	 *
+	 * @return array Array of $state_slug => $state_label pairs.
+	 */
+	public function get_site_states( $site ) {
+		$available_states = array(
+			'public'   => __( 'public', 'wordpress-seo' ),
+			'archived' => __( 'archived', 'wordpress-seo' ),
+			'mature'   => __( 'mature', 'wordpress-seo' ),
+			'spam'     => __( 'spam', 'wordpress-seo' ),
+			'deleted'  => __( 'deleted', 'wordpress-seo' ),
+		);
+
+		$site_states = array();
+		foreach ( $available_states as $state_slug => $state_label ) {
+			if ( $site->$state_slug === '1' ) {
+				$site_states[ $state_slug ] = $state_label;
+			}
+		}
+
+		return $site_states;
+	}
+
+	/**
+	 * Handles a request to update plugin network options.
+	 *
+	 * This method works similar to how option updates are handled in `wp-admin/options.php` and
+	 * `wp-admin/network/settings.php`.
+	 *
+	 * @return void
+	 */
+	public function handle_update_options_request() {
+		$option_group = filter_input( INPUT_POST, 'network_option_group', FILTER_SANITIZE_STRING );
+
+		$this->verify_request( "$option_group-network-options" );
```

Comment:
When using a variable in a string like this, I'd really prefer to wrap it with braces, or concat the string.
`"${option_group}-network-options"`